apps/session-manager/src/services/redis-session-storage.ts

```diff
@@ -0,0 +1,684 @@
+            .setEx(
+              `${myPortalTokenPrefix}${user.myportal_token}`,
+              expireSec,
+              user.session_token,
+            ),
+        E.toError,
+      ),
+      singleStringReplyAsync,
+      falsyResponseToErrorAsync(new Error("Error setting MyPortal token")),
+    );
+
+    const setBPDTokenV2 = pipe(
+      TE.tryCatch(
+        () =>
+          redisClientSelector
+            .selectOne(RedisClientMode.FAST)
+            .setEx(
+              `${bpdTokenPrefix}${user.bpd_token}`,
+              expireSec,
+              user.session_token,
+            ),
+        E.toError,
+      ),
+      singleStringReplyAsync,
+      falsyResponseToErrorAsync(new Error("Error setting BPD token")),
+    );
+
+    const setZendeskTokenV2 = pipe(
+      TE.tryCatch(
+        () =>
+          redisClientSelector
+            .selectOne(RedisClientMode.FAST)
+            .setEx(
+              `${zendeskTokenPrefix}${user.zendesk_token}`,
+              expireSec,
+              user.session_token,
+            ),
+        E.toError,
+      ),
+      singleStringReplyAsync,
+      falsyResponseToErrorAsync(new Error("Error setting Zendesk token")),
+    );
+
+    const setFIMSTokenV2 = pipe(
+      TE.tryCatch(
+        () =>
+          redisClientSelector
+            .selectOne(RedisClientMode.FAST)
+            .setEx(
+              `${fimsTokenPrefix}${user.fims_token}`,
+              expireSec,
+              user.session_token,
+            ),
+        E.toError,
+      ),
+      singleStringReplyAsync,
+      falsyResponseToErrorAsync(new Error("Error setting FIMS token")),
+    );
+
+    // If is a session update, the session info key doesn't must be updated.
+    // eslint-disable-next-line functional/no-let
+    let saveSessionInfoPromise: TE.TaskEither<Error, boolean> = TE.right(true);
+    if (!isUserSessionUpdate) {
+      const sessionInfo: SessionInfo = {
+        createdAt: new Date(),
+        sessionToken: user.session_token,
+      };
+      saveSessionInfoPromise = saveSessionInfo(redisClientSelector)(
+        sessionInfo,
+        user.fiscal_code,
+        expireSec,
+      );
+    }
+
+    const removeOtherUserSessionsPromise =
+      removeOtherUserSessions(redisClientSelector)(user);
+
+    return pipe(
+      A.sequence(T.taskSeq)([
+        setSessionTokenV2,
+        setWalletTokenV2,
+        setMyPortalTokenV2,
+        setBPDTokenV2,
+        setZendeskTokenV2,
+        setFIMSTokenV2,
+        saveSessionInfoPromise,
+        pipe(
+          TE.tryCatch(() => removeOtherUserSessionsPromise, E.toError),
+          TE.mapLeft((e) => E.left<Error, boolean>(e)),
+          TE.toUnion,
+        ),
+      ]),
+      TE.fromTask,
+      TE.chain(
+        TE.fromPredicate(A.some(E.isLeft), (results) =>
+          multipleErrorsFormatter(
+            results.filter(E.isLeft).map((result) => result.left),
+            "RedisSessionStorage.set",
+          ),
+        ),
```

Comment:
with the code you provided we would have reached the left part if the array **didn't** contain `Either.left`.
```suggestion
                TE.fromPredicate(
          (errorOrResults) => !A.some(E.isLeft)(errorOrResults),
          (results) =>
            multipleErrorsFormatter(
              results.filter(E.isLeft).map((result) => result.left),
              "RedisSessionStorage.set",
            ),
        ),
```

Reply:
Fixed in [c695c50](https://github.com/pagopa/io-auth-n-identity-domain/pull/10/commits/c695c50d15cfd6d2e3935ae5c770a8f9ceb650a2)